### PR TITLE
CM-185: Required Parameters in Blazor

### PR DESCRIPTION
**Pull request by semirteskeredzic**

### CODE REVIEW

File: blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazorTests/UnitTest1.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using RequiredParametersInBlazor;
+using RequiredParametersInBlazor.Components;
+
+namespace RequiredParametersInBlazor.Tests
+{
+    public class AgeCardTests : TestContext
+    {
+        [Fact]
+        public void WhenYearOfBirthIsSupplied_ThenAgeIsReturned()
```
Comment (PhilBroderick) [inline blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazorTests/UnitTest1.cs:9]: Should be `Given_When_Then` format.

File: blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazor/Components/AgeCard.razor [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    // Demonstrates how a parent component can supply required parameters
+    [Parameter, EditorRequired]
+    public int YearOfBirthInput { get; set; }
+
+    // Method that calculates current age from the supplied year of birth
+    public int GetAge()
+    {
+        DateTime currentYear = DateTime.Now;
+        int year = currentYear.Year;
+        int age = year - YearOfBirthInput;
```
[the remark this answers is not in the document]
Comment (semirteskeredzic) [reply blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazor/Components/AgeCard.razor]: Removed explicit data type, changed all three to var

File: blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazor/Program.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using Microsoft.AspNetCore.Components.Web;
+using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
+using RequiredParametersInBlazor;
+
+var builder = WebAssemblyHostBuilder.CreateDefault(args);
+builder.RootComponents.Add<App>("#app");
+builder.RootComponents.Add<HeadOutlet>("head::after");
+
+builder.Services.AddScoped(sp => new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });
```
[the remark this answers is not in the document]
Comment (semirteskeredzic) [reply blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazor/Program.cs]: Line is split after instantiating new HttpClient

File: blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazorTests/UnitTest1.cs (hunk 1 above)
Comment (semirteskeredzic) [reply blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazorTests/UnitTest1.cs]: Implemented Given_When_Then for the method name

File: blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazor/Properties/launchSettings.json [hunk 4]
```diff
@@ -17,6 +9,24 @@
         "ASPNETCORE_ENVIRONMENT": "Development"
       },
       "dotnetRunMessages": true
+    },
```
Comment (PhilBroderick) [inline blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazor/Properties/launchSettings.json:12]: These were removed and then added back again in error.

File: blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazorTests/UnitTest1.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿using RequiredParametersInBlazor;
```
Comment (PhilBroderick) [inline blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazorTests/UnitTest1.cs:1]: Just noticed that this file is called `UnitTest1`. Instead it should be `AgeCardTests`.

File: blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazorTests/UnitTest1.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+            var age = ageCard.Instance.GetAge();
+
+            // Assert
+            var expectedAge = GetCurrentYear() - yearOfBirthInput;
+            Assert.Equal(expectedAge, age);
+        }
+
+        private int GetCurrentYear()
+        {
+            return 2023;
```
Comment (PhilBroderick) [inline blazor-features/RequiredParametersInBlazor/RequiredParametersInBlazorTests/UnitTest1.cs:27]: Does this need to be hardcoded? Can't we just use `DateTime.UtcNow.Year`?

**Review by PhilBroderick — CHANGES_REQUESTED** [review]

**Review by PhilBroderick — APPROVED** [review]

**Pull request merged**